swagger.yml

```diff
@@ -214,3 +220,238 @@ paths:
+
+  /v2/task/frameworks/<framework-id>/executors/<executor-id>/runs/<container-id>/<file>:
+    get:
+      description: |
+          Read the SINGLE task log. framework-id, executor-id, container-id and file are required task unique parameters. Available on agent nodes.
+      parameters:
+        - $ref: "#/parameters/filter"
+        - $ref: "#/parameters/limit"
+        - $ref: "#/parameters/skip"
+        - $ref: "#/parameters/cursor"
+      responses:
+        200:
+          description: Successful response.
+        400:
+          description: Bad request, field not found.
+        401:
+          description: Not authorized.
+        500:
+          description: Internal server error.
+
+  /v2/task/frameworks/<framework-id>/executors/<executor-id>/runs/<container-id>/<file>/download:
+    get:
+      description: |
+          Download the SINGLE task log. framework-id, executor-id, container-id and file are required task unique parameters. Available on agent nodes.
+      responses:
+        200:
+          description: Successful response.
+        400:
+          description: Bad request, field not found.
+        401:
+          description: Not authorized.
+        500:
+          description: Internal server error.
+
+  /v2/task/<task-id>/file/<file_name>:
+      get:
+        description: |
+            This endpoint discovers the metadata assosiated with a given <task-id> and redirects to a specific endpoint on agent node. Reading <file_name> file.
+        parameters:
+        - $ref: "#/parameters/filter"
+        - $ref: "#/parameters/limit"
+        - $ref: "#/parameters/skip"
+        - $ref: "#/parameters/cursor"
+        responses:
+          200:
+            description: Successful response.
+          400:
+            description: Bad request, field not found.
+          401:
+            description: Not authorized.
+          500:
+            description: Internal server error.
+
+  /v2/task/<task-id>/browse:
+      get:
+        description: |
+            This endpoint discovers the metadata assosiated with a given <task-id> and redirects to a specific browsing endpoint on agent node.
+        responses:
+          200:
+            description: Successful response.
+          400:
+            description: Bad request, field not found.
+          401:
+            description: Not authorized.
+          500:
+            description: Internal server error.
+
+  /v2/task/<task-id>/download:
+      get:
+        description: |
+            Download stdout for a given <task-id>
+        responses:
+          200:
+            description: Successful response.
+          400:
+            description: Bad request, field not found.
+          401:
+            description: Not authorized.
+          500:
+            description: Internal server error.
+
+  /v2/task/<task-id>/file/<file_name>/download:
+      get:
+        description: |
+            Download <file_name> for a given <task-id>
```

Comment:
Why is `file_name` using underscores here?  Elsewhere we use `-` to separate path variables.